iwyu_ast_util.cc

```diff
@@ -1299,12 +1325,20 @@ GetTplTypeResugarMapForClassNoComponentTypes(const clang::Type* type) {
     return retval;
 
   // Get the list of template args that apply to the decls.
-  const NamedDecl* decl = TypeToDeclForContent(tpl_spec_type);
-  const ClassTemplateSpecializationDecl* tpl_decl = DynCastFrom(decl);
-  if (!tpl_decl)   // probably because tpl_spec_type is a dependent type
-    return retval;
-  const TemplateArgumentList& tpl_args
-      = tpl_decl->getTemplateInstantiationArgs();
+  const NamedDecl* decl = TypeToDeclAsWritten(tpl_spec_type);
```

Comment:
I'd like to see this move down to just before its use.

Reply:
Code rearranged.